### PR TITLE
MED-338 dockerhub implementation

### PR DESCRIPTION
**Pull request by harshal359**

### CODE REVIEW

**Comment by rssanders3** [comment]
Note: Ensure the name includes the literal Jira ticket: Med 338 -> MED-338

File: nextflow_aws_batch_trigger.py [hunk 1]
```diff
@@ -19,23 +19,26 @@ def create_launch_template(
...
-                    sudo s3fs {s3_bucket} {s3fs_mount} -o allow_other -o umask=000 -o iam_role=auto
-
-                    --==MYBOUNDARY==--"""
+        user_data = (
+            """MIME-Version: 1.0\nContent-Type: multipart/mixed; boundary="==MYBOUNDARY=="\n\n--==MYBOUNDARY==\n"""
+            """Content-Type: text/cloud-config; charset="us-ascii"\n\npackages:\n- jq\n- aws-cli\n\n"""
+            """runcmd:\n"""
+            """- amazon-linux-extras install epel -y\n"""
+            """- yum install s3fs-fuse -y\n"""
+            """- /usr/bin/aws configure set region $(curl http://169.254.169.254/latest/meta-data/placement/region)\n"""
```
Comment (rssanders3) [inline nextflow_aws_batch_trigger.py]: Instead of calling the url listed, use the `region_name` we pass in.

**Review by rssanders3 — CHANGES_REQUESTED** [review]

File: nextflow_aws_batch_trigger.py (hunk 1 above)
Comment (harshal359) [reply nextflow_aws_batch_trigger.py]: have passed variable 'region_name' now on line 28.

File: nextflow_aws_batch_trigger.py [hunk 2]
```diff
@@ -19,23 +19,26 @@ def create_launch_template(
...
-                    sudo amazon-linux-extras install epel -y
-                    sudo yum install s3fs-fuse -y
-                    mkdir {s3fs_mount}
-                    mkdir {result_location}
-                    chmod 777 {s3fs_mount}
-                    sudo s3fs {s3_bucket} {s3fs_mount} -o allow_other -o umask=000 -o iam_role=auto
-
-                    --==MYBOUNDARY==--"""
+        user_data = (
+            """MIME-Version: 1.0\nContent-Type: multipart/mixed; boundary="==MYBOUNDARY=="\n\n--==MYBOUNDARY==\n"""
```
Comment (rssanders3) [inline nextflow_aws_batch_trigger.py]: Change to single strings (")

File: nextflow_aws_batch_trigger.py [hunk 3]
```diff
@@ -19,23 +19,26 @@ def create_launch_template(
...
+            """Content-Type: text/cloud-config; charset="us-ascii"\n\npackages:\n- jq\n- aws-cli\n\n"""
+            """runcmd:\n"""
+            """- amazon-linux-extras install epel -y\n"""
+            """- yum install s3fs-fuse -y\n"""
+            f"""- /usr/bin/aws configure set region {region_name}\n"""
```
Comment (rssanders3) [inline nextflow_aws_batch_trigger.py]: region_name added in function args

**Review by rssanders3 — CHANGES_REQUESTED** [review]

**Review by rssanders3 — APPROVED** [review]

**Pull request merged**